extension/llm/runner/runner_interface.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// An interface for LLM runners. Developers can create their own runner that
+// implements their own load and generation logic to run the model.
+
+#pragma once
+
+#include <functional>
+#include <string>
+
+#include <executorch/extension/llm/runner/stats.h>
+#include <executorch/extension/module/module.h>
+
+namespace executorch {
+namespace extension {
+namespace llm {
+
+class ET_EXPERIMENTAL RunnerInterface {
```

Comment:
Maybe call it `IRunner`? Also what is the actual usage of having this interface? Can you also include the changes you want to make, to reduce duplicate code?

Reply:
> Maybe call it IRunner
Works for me

Actual usage: https://github.com/pytorch/executorch/pull/6208/files#diff-344496b5b78dc04421b5eddcb0f261269c3150e122a9be63676da565955cb4f7R123-R131

That code is not ready yet